pkgs/development/compilers/circt/default.nix

```diff
@@ -0,0 +1,59 @@
+{ stdenv
+, lib
+, cmake
+, coreutils
+, python3
+, git
+, fetchFromGitHub
+, ninja
+}:
+
+let
+  pname = "circt";
+  version = "1.29.0";
+  src = fetchFromGitHub {
+    owner = "llvm";
+    repo = "circt";
+    rev = "firtool-${version}";
+    sha256 = "sha256-HsXwh98RZuXvK/KkZ2NAGwWNLxUAQVj+WKzZXd4C4Is=";
+    fetchSubmodules = true;
+  };
+  psutil = python3.withPackages (ps: [ ps.psutil ]);
+in
+  stdenv.mkDerivation {
+    inherit version src pname;
+    nativeBuildInputs = [ cmake ninja git psutil ];
+
+    cmakeDir = "../llvm/llvm";
+    cmakeFlags = [
+      "-DLLVM_ENABLE_BINDINGS=OFF"
+      "-DLLVM_ENABLE_OCAMLDOC=OFF"
+      "-DLLVM_BUILD_EXAMPLES=OFF"
+      "-DLLVM_OPTIMIZED_TABLEGEN=ON"
+      "-DLLVM_ENABLE_PROJECTS=mlir"
+      "-DLLVM_EXTERNAL_PROJECTS=circt"
+      "-DLLVM_EXTERNAL_CIRCT_SOURCE_DIR=.."
+      "-DCIRCT_LLHD_SIM_ENABLED=OFF"
+    ];
+
+    preConfigure = ''
+      substituteInPlace test/circt-reduce/test/annotation-remover.mlir --replace "/usr/bin/env" "${coreutils}/bin/env"
```

Comment:
use `patchShebangs`

Reply:
This is not a shebang, but a [hardcoded path in test spec](https://github.com/llvm/circt/blob/main/test/circt-reduce/test/annotation-remover.mlir#L3). `patchShebangs` seems not working for this?